src/hotspot/os/linux/cgroupV2Subsystem_linux.hpp

```diff
@@ -28,38 +28,8 @@
 #include "cgroupSubsystem_linux.hpp"
 
 class CgroupV2Controller: public CgroupController {
-  private:
-    /* the mount path of the cgroup v2 hierarchy */
-    char *_mount_path;
-    /* The cgroup path for the controller */
-    char *_cgroup_path;
-    bool _read_only;
-
-    /* Constructed full path to the subsystem directory */
-    char *_path;
-    static char* construct_path(char* mount_path, char *cgroup_path);
-
   public:
-    CgroupV2Controller(char* mount_path,
-                       char *cgroup_path,
-                       bool ro) :  _mount_path(os::strdup(mount_path)),
-                                   _cgroup_path(os::strdup(cgroup_path)),
-                                   _read_only(ro),
-                                   _path(construct_path(mount_path, cgroup_path)) {
-    }
-    // Shallow copy constructor
```

Comment:
I had to remove the "shallow" part as it was not memory safe afterwards and one could easily get:
```
=================================================================
==3770369==ERROR: AddressSanitizer: heap-use-after-free on address 0x502000001530 at pc 0x7f086bcc45c0 bp 0x7f086adfd7b0 sp 0x7f086adfcf70
READ of size 1 at 0x502000001530 thread T1
    #0 0x7f086bcc45bf in strcmp.part.0 (/lib64/libasan.so.8+0xc45bf) (BuildId: c1431025b5d8af781c22c9ceea71f065c547d32d)
    #1 0x7f0862c77e55 in CgroupController::set_path(char const*) /home/azul/azul/openjdk-git/src/hotspot/os/linux/cgroupSubsystem_linux.cpp:841
...
0x502000001530 is located 0 bytes inside of 2-byte region [0x502000001530,0x502000001532)
freed by thread T1 here:
...
    #4 0x7f08655c537c in os::free(void*) /home/azul/azul/openjdk-git/src/hotspot/share/runtime/os.cpp:775
    #5 0x7f0862c7a456 in CgroupController::~CgroupController() /home/azul/azul/openjdk-git/src/hotspot/os/linux/cgroupSubsystem_linux.hpp:184
...     
previously allocated by thread T1 here:
    #0 0x7f086bcf7997 in malloc (/lib64/libasan.so.8+0xf7997) (BuildId: c1431025b5d8af781c22c9ceea71f065c547d32d)
...     
    #7 0x7f08655c87df in os::strdup(char const*, MEMFLAGS) /home/azul/azul/openjdk-git/src/hotspot/share/runtime/os.cpp:605
    #8 0x7f0862c7a26f in CgroupController::CgroupController(char*, char*, bool) /home/azul/azul/openjdk-git/src/hotspot/os/linux/cgroupSubsystem_linux.hpp:170
```
I plan to fix the ugly value-copy in a later refactorization.